From ab95684623120c3f1a95b9030109bb6afe2280f5 Mon Sep 17 00:00:00 2001
From: Angela Stegmaier <angelabaker@ti.com>
Date: Tue, 20 Aug 2019 15:59:02 -0500
Subject: [PATCH] v4l2object: Update formats table to include YUV422
 multiplanar

YUV422 multiplanar is missing from the table. Add the support
for this format.

Upstream-Status: Pending

Signed-off-by: Angela Stegmaier <angelabaker@ti.com>

---
 sys/v4l2/gstv4l2object.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 6fd3080..2d401a7 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -130,6 +130,7 @@ static const GstV4L2FormatDesc gst_v4l2_formats[] = {
   {V4L2_PIX_FMT_UYVY, TRUE, GST_V4L2_RAW},
   {V4L2_PIX_FMT_VYUY, TRUE, GST_V4L2_RAW},
   {V4L2_PIX_FMT_YUV422P, TRUE, GST_V4L2_RAW},
+  {V4L2_PIX_FMT_YUV422M, TRUE, GST_V4L2_RAW},
   {V4L2_PIX_FMT_YUV411P, TRUE, GST_V4L2_RAW},
   {V4L2_PIX_FMT_Y41P, TRUE, GST_V4L2_RAW},
   {V4L2_PIX_FMT_YUV444, TRUE, GST_V4L2_RAW},
@@ -1090,6 +1091,7 @@ gst_v4l2_object_format_get_rank (const struct v4l2_fmtdesc *fmt)
       rank = YUV_BASE_RANK + 4;
       break;
     case V4L2_PIX_FMT_YUV422P: /* Y42B, 16 bits per pixel */
+    case V4L2_PIX_FMT_YUV422M:
       rank = YUV_BASE_RANK + 8;
       break;
 
@@ -1345,6 +1347,7 @@ gst_v4l2_object_v4l2fourcc_to_video_format (guint32 fourcc)
       format = GST_VIDEO_FORMAT_Y41B;
       break;
     case V4L2_PIX_FMT_YUV422P:
+    case V4L2_PIX_FMT_YUV422M:
       format = GST_VIDEO_FORMAT_Y42B;
       break;
     case V4L2_PIX_FMT_YVYU:
@@ -1493,6 +1496,7 @@ gst_v4l2_object_v4l2fourcc_to_bare_struct (guint32 fourcc)
     case V4L2_PIX_FMT_YVU420:
     case V4L2_PIX_FMT_UYVY:
     case V4L2_PIX_FMT_YUV422P:
+    case V4L2_PIX_FMT_YUV422M:
     case V4L2_PIX_FMT_YVYU:
     case V4L2_PIX_FMT_YUV411P:{
       GstVideoFormat format;
@@ -1723,6 +1727,7 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
         break;
       case GST_VIDEO_FORMAT_Y42B:
         fourcc = V4L2_PIX_FMT_YUV422P;
+        fourcc_nc = V4L2_PIX_FMT_YUV422M;
         break;
       case GST_VIDEO_FORMAT_NV12:
         fourcc = V4L2_PIX_FMT_NV12;
